app/controllers/image_search_controller.rb

```diff
@@ -4,6 +4,12 @@ class ImageSearchController < ::ApplicationController
   def auto_complete_repository_name
     render :text => (use_hub? ? hub_image_exists?(params[:search]) :
         registry_image_exists?(params[:search])).to_s
+  rescue Docker::Error::NotFoundError => e
```

Comment:
Make a helper method in which you just yield the core functionality as a block, similar to what's in here https://github.com/theforeman/foreman-docker/blob/master/app/models/foreman_docker/docker.rb#L116 

Otherwise I think it's going to be a pain every time we have to change any rescue.